services/frontend-service/pkg/cmd/server.go

```diff
@@ -313,32 +313,50 @@ func runServer(ctx context.Context) error {
 	httpHandler := handler.Server{
 		BatchClient:   batchClient,
 		RolloutClient: rolloutClient,
+		VersionClient: api.NewVersionServiceClient(cdCon),
 		Config:        c,
 		KeyRing:       pgpKeyRing,
 		AzureAuth:     c.AzureEnableAuth,
 	}
 	mux := http.NewServeMux()
-	mux.Handle("/environments/", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+	restHandler := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		defer readAllAndClose(req.Body, 1024)
 		if c.DexEnabled {
-			interceptors.DexLoginInterceptor(w, req, httpHandler, c.DexClientId, c.DexClientSecret)
+			interceptors.DexLoginInterceptor(w, req, httpHandler.Handle, c.DexClientId, c.DexClientSecret)
+			return
```

Comment:
```suggestion
```
This `return` would mean that we exit whenever dex is enabled.

Reply:
Which we should do, right? Because the `DexLoginInterceptor` [calls the underlying http handler in any case](https://github.com/freiheit-com/kuberpult/blob/006f7b4ec1cc0422416f89cc9467159c1f3b6397/services/frontend-service/pkg/interceptors/interceptors.go#L116) so there is no point calling it again.

Alternatively, we could just return the modified context and then let the http handler do the validation of correct headers.

Reply:
Discussed in person: We keep the return.